### PR TITLE
Re enable machineinfo test

### PR DESCRIPTION
**Pull request by marioferh**
Re-enable machineinfo test with new updated data of [#59 ](https://github.com/openshift-kni/debug-tools/pull/59)
Add clean of CPUVendorID to cleanProcfs to avoid system dependence
Remove -P from machineinfo command. Not used procfs in machineinfotest and can lead to confusion

### CODE REVIEW

**Comment by marioferh** [comment]
should I change it even if arch dependent?
`         - 	"vendor_id": string("AuthenticAMD"),
        + 	"vendor_id": string("GenuineIntel"), `

**Comment by fromanirh** [comment]
/lgtm
again extremely terse commit message (for example: reference the PR updating the e2e data; declaring that thanks to that we can now re-enable the tests), but good enough for now.

**Comment by yanirq** [comment]
Did you rebase on top of https://github.com/openshift-kni/debug-tools/pull/59  ?  I see one test failing

**Comment by marioferh** [comment]
> Did you rebase on top of #59 ? I see one test failing

Yep, rebase it, but its failing, checking it

**Comment by marioferh** [comment]
there is some info in the machineinfo arch dependent, no all date comes from /proc /sys

**Comment by fromanirh** [comment]
> there is some info in the machineinfo arch dependent, no all date comes from /proc /sys

which infos?

**Comment by marioferh** [comment]
I can see many differences between run in the same machine as generated json and in the ci:

e.g.:
```
     - 	"memory_by_type": map[string]interface{}{},
    + 	"memory_by_type": map[string]interface{}{
    + 		"Unbuffered-DDR4": map[string]interface{}{"capacity": float64(1.03079215104e+11), "dimm_count": float64(12)},
    + 	}, 
```

**Comment by fromanirh** [comment]
or, if the data is not relevant for us, let's make sure we never collect it: https://github.com/openshift-kni/debug-tools/blob/main/pkg/machineinformer/informer.go#L49

**Comment by marioferh** [comment]
> or, if the data is not relevant for us, let's make sure we never collect it: https://github.com/openshift-kni/debug-tools/blob/main/pkg/machineinformer/informer.go#L49

I have to investigate which data is relevant

File: test/data/dell_2_numa/machineinfo.json [hunk 1]
```diff
@@ -1 +1,1565 @@
...
+{
+    "timestamp": "2000-01-01T00:00:00Z",
+    "vendor_id": "AuthenticAMD",
```
Comment (fromanirh) [inline test/data/dell_2_numa/machineinfo.json:3]: from where does this come from?

**Review by fromanirh — COMMENTED** [review]
typo: should read `machineinfo`

**Comment by marioferh** [comment]
@fromanirh there are three differences in machineinfo.json
"vendor_id": "AuthenticAMD",
"num_cores": 104,
this can be hardcoded

And the topology that is different in local cluster and ci cluster.

I don't know if can be hard-coded or not

File: test/data/dell_2_numa/machineinfo.json (hunk 1 above)
Comment (marioferh) [reply test/data/dell_2_numa/machineinfo.json:3]: its arch dependent, it comes from the ci cluster.

**Comment by marioferh** [comment]
> typo: should read `machineinfo`

sorry, where?

**Comment by fromanirh** [comment]
> > typo: should read `machineinfo`
> 
> sorry, where?

in the commit message proper
image

File: test/data/dell_2_numa/machineinfo.json (hunk 1 above)
Comment (fromanirh) [reply test/data/dell_2_numa/machineinfo.json:3]: I'm pretty sure the cluster is Intel, though

**Comment by fromanirh** [comment]
before you submit the final version, note you still have the `macineinfo` typo in the commit message

**Comment by marioferh** [comment]
> before you submit the final version, note you still have the `macineinfo` typo in the commit message

already fixed

**Comment by marioferh** [comment]
@fromanirh fixed it.
The initial issue was: tests launched in local extract info and then delete it, when I generate the output json with local command it was taking the local sys, not the uncompressed one from the tar :S
The second was, CPUVendorID was not cleaned with the --clean-procfs-info. I've already added that cleanup. 
The command has -P ./test/data/dell_2_numa/proc then I though that al proc data comes from local folder not from system proc, that confused me.

File: pkg/machineinformer/informer.go [hunk 2]
```diff
@@ -50,6 +50,7 @@ func (handle *Handle) Run() {
 		info.NumPhysicalCores = 0
 		info.CpuFrequency = 0
 		info.MemoryCapacity = 0
+		info.CPUVendorID = ""
```
Comment (fromanirh) [inline pkg/machineinformer/informer.go:53]: this..

File: test/e2e/knit_machineinfo_test.go [hunk 3]
```diff
@@ -21,11 +21,9 @@ var _ = g.Describe("knit machineinfo tests", func() {
 
 	g.Context("With isolated, reserved CPUs", func() {
 		g.It("Produces the expected machineinfo output", func() {
-			g.Skip("broken until we refresh the machineinfo data")
 
 			cmdline := []string{
 				filepath.Join(binariesPath, "knit"),
-				"-P", filepath.Join(snapshotRoot, "proc"),
```
Comment (fromanirh) [inline test/e2e/knit_machineinfo_test.go:28]: ...combined to this is not a good direction. We should have stronger isolation from the system, not weaker. And we should minimize the information we ignore from the virtual filesystem, not ignoring more.

It seems to me that the core issue here is that `knit` doesn't have a recommended way or flow to collect sysinfo.tgz, thus the content of the provided archive is easily out of sync, causing all kind issues.
We need to fix this step before, and/or document clearly (few lines in the README or in another README are fine) how the update process of `test/data` is supposed to look.

**Comment by fromanirh** [comment]
ok, this is more complex than we initially thought. We can pause this work and resume later when there is more bandwidth and we can sort out and tackle the dependencies.

File: test/e2e/knit_machineinfo_test.go (hunk 3 above)
Comment (marioferh) [reply test/e2e/knit_machineinfo_test.go:28]: I think is a misunderstanding here, machineinfo is not taking local files with -P: 
`➜  debug-tools git:(re_enable_machineinfo_test) ✗ grep -Rni "knitOpts.ProcFSRoot"            
pkg/knit/cmd/ghw/ghw.go:93:                     "/proc": knitOpts.ProcFSRoot,
pkg/knit/cmd/cpuaff.go:67:      ph := procs.New(knitOpts.Log, knitOpts.ProcFSRoot)
pkg/knit/cmd/cpuaff.go:88:              return fmt.Errorf("error getting process infos from %q: %v", knitOpts.ProcFSRoot, err)
pkg/knit/cmd/irqaff.go:77:      ih := irqs.New(knitOpts.Log, knitOpts.ProcFSRoot)
pkg/knit/cmd/irqaff.go:86:              return fmt.Errorf("error parsing irqs from %q: %v", knitOpts.ProcFSRoot, err)
pkg/knit/cmd/irqaff.go:116:     sh := softirqs.New(knitOpts.Log, knitOpts.ProcFSRoot)
pkg/knit/cmd/irqaff.go:120:             return fmt.Errorf("error parsing softirqs from %q: %v", knitOpts.ProcFSRoot, err)
pkg/knit/cmd/root.go:77:        root.PersistentFlags().StringVarP(&knitOpts.ProcFSRoot, "procfs", "P", "/proc", "procfs root")
pkg/knit/cmd/irqwatch.go:66:    ih := irqs.New(knitOpts.Log, knitOpts.ProcFSRoot)
`

Comment (marioferh) [reply test/e2e/knit_machineinfo_test.go:28]: that is the reason why that file has been deleted and info.CPUVendorID = ""

Comment (fromanirh) [reply test/e2e/knit_machineinfo_test.go:28]: thing is, `knit` *should* consume the local files from the path given using `-P`. The combined action of `-P` and `-S` is supposed to fully isolate `knit` from host procfs and sysfs. The `--clean-procfs-info` is already too heavy handed, and arguably half a hack in the first place. Improving this part is out of scope from this PR; however, we can and should not making it any worse. This is why I think the current direction -getting rid of `-P` entirely- is not ideal.

Comment (marioferh) [reply test/e2e/knit_machineinfo_test.go:28]: ok, understood

**Pull request closed without merging**